evm/logic/call.py

```diff
@@ -342,3 +342,101 @@ def compute_msg_extra_gas(self, computation, gas, to, value):
+
+        call_data = computation.memory.read(memory_input_start_position, memory_input_size)
+
+        #
+        # Message gas allocation and fees
+        #
+        child_msg_gas, child_msg_gas_fee = self.compute_msg_gas(computation, gas, to, value)
+        computation.gas_meter.consume_gas(child_msg_gas_fee, reason=self.mnemonic)
+
+        # Pre-call checks
+        with computation.state_db(read_only=True) as state_db:
+            sender_balance = state_db.get_balance(computation.msg.storage_address)
+
+        insufficient_funds = should_transfer_value and sender_balance < value
+        stack_too_deep = computation.msg.depth + 1 > constants.STACK_DEPTH_LIMIT
+
+        if insufficient_funds or stack_too_deep:
+            computation.return_data = b''
+            if insufficient_funds:
+                err_message = "Insufficient Funds: have: {0} | need: {1}".format(
+                    sender_balance,
+                    value,
+                )
+            elif stack_too_deep:
+                err_message = "Stack Limit Reached"
+            else:
+                raise Exception("Invariant: Unreachable code path")
+
+            self.logger.debug(
+                "%s failure: %s",
+                self.mnemonic,
+                err_message,
+            )
+            computation.gas_meter.return_gas(child_msg_gas)
+            computation.stack.push(0)
+        else:
+            with computation.state_db(read_only=True) as state_db:
+                if code_address:
+                    code = state_db.get_code(code_address)
+                else:
+                    code = state_db.get_code(to)
+
+            child_msg_kwargs = {
+                'gas': child_msg_gas,
+                'value': value,
+                'to': to,
+                'data': call_data,
+                'code': code,
+                'transaction_gas_limit': computation.msg.transaction_gas_limit,
+                'code_address': code_address,
+                'should_transfer_value': should_transfer_value,
+                'is_static': is_static,
+            }
+            if sender is not None:
+                child_msg_kwargs['sender'] = sender
+
+            child_msg = computation.prepare_child_message(**child_msg_kwargs)
+
+            child_computation = computation.apply_child_computation(child_msg)
+
+            if child_computation.is_error:
+                computation.stack.push(0)
+            else:
+                computation.stack.push(1)
+
+            if not child_computation.should_erase_return_data:
+                actual_output_size = min(memory_output_size, len(child_computation.output))
+                computation.memory.write(
+                    memory_output_start_position,
+                    actual_output_size,
+                    child_computation.output[:actual_output_size],
+                )
+
+            if not child_computation.should_burn_gas:
```

Comment:
Note to self.

this is a confusing `if` statement because of the need for the `not`.  We might make a second property which returns the inverse of this as `computation.should_return_gas`.

Reply:
Noted. Will make a PR to update this.